File: Introduccion_a_POO/Ej05_Persona/Persona.py
class Persona:
    def __init__(self, nombre, paterno, materno, edad, ci):
        self.nombre = nombre
        self.paterno = paterno
        self.materno = materno
        self.edad = edad
        self.ci = ci
    def mostrarDatos(self):
        print(f"--- Datos de la Persona ---")
        print(f"Nombre completo: {self.nombre} {self.paterno} {self.materno}")
        print(f"Edad: {self.edad}")
        print(f"CI: {self.ci}")
        print(f"---------------------------")
    def esMayorDeEdad(self):
        return self.edad >= 18
    def modificarEdad(self, nuevaEdad):
        if nuevaEdad > 0:
            self.edad = nuevaEdad
        else:
            print("Edad inválida.")
    def getPaterno(self):
        return self.paterno
if __name__ == '__main__':
    persona1 = Persona("Ana", "Quispe", "Mendoza", 25, "1234567 LP")
    persona2 = Persona("Carlos", "Quispe", "Rojas", 17, "7654321 CB")
    print("Instancias creadas:")
    persona1.mostrarDatos()
    persona2.mostrarDatos()
    print(f"\n¿{persona1.nombre} es mayor de edad? {'Sí' if persona1.esMayorDeEdad() else 'No'}")
    print(f"¿{persona2.nombre} es mayor de edad? {'Sí' if persona2.esMayorDeEdad() else 'No'}")
    print("\nModificando la edad de Carlos a 18...")
    persona2.modificarEdad(18)
    print(f"¿{persona2.nombre} es ahora mayor de edad? {'Sí' if persona2.esMayorDeEdad() else 'No'}")
    persona2.mostrarDatos()
    print("\nVerificando apellidos paternos...")
    if persona1.getPaterno() == persona2.getPaterno():
        print("Las dos personas tienen el mismo apellido paterno.")
    else:
        print("Las dos personas tienen apellidos paternos diferentes.")

File: Introduccion_a_POO/Ej05_Persona/Persona.java
package Introduccion_a_POO.Ej05_Persona;
public class Persona {
    private String nombre;
    private String paterno;
    private String materno;
    private int edad;
    private String ci;
    public Persona(String nombre, String paterno, String materno, int edad, String ci) {
        this.nombre = nombre;
        this.paterno = paterno;
        this.materno = materno;
        this.edad = edad;
        this.ci = ci;
    }
    public void mostrarDatos() {
        System.out.println("--- Datos de la Persona ---");
        System.out.println("Nombre completo: " + nombre + " " + paterno + " " + materno);
        System.out.println("Edad: " + edad);
        System.out.println("CI: " + ci);
        System.out.println("---------------------------");
    }
    public boolean esMayorDeEdad() {
        return this.edad >= 18;
    }
    public void modificarEdad(int nuevaEdad) {
        if (nuevaEdad > 0) {
            this.edad = nuevaEdad;
        } else {
            System.out.println("Edad inválida.");
        }
    }
    public String getPaterno() {
        return this.paterno;
    }   
    public String getNombre() {
        return this.nombre;
    }
    public static void main(String[] args) {
        Persona persona1 = new Persona("Ana", "Quispe", "Mendoza", 25, "1234567 LP");
        Persona persona2 = new Persona("Carlos", "Quispe", "Rojas", 17, "7654321 CB");
        System.out.println("Instancias creadas:");
        persona1.mostrarDatos();
        persona2.mostrarDatos();
        System.out.println("\n¿" + persona1.getNombre() + " es mayor de edad? " + (persona1.esMayorDeEdad() ? "Sí" : "No"));
        System.out.println("¿" + persona2.getNombre() + " es mayor de edad? " + (persona2.esMayorDeEdad() ? "Sí" : "No"));
        System.out.println("\nModificando la edad de Carlos a 18...");
        persona2.modificarEdad(18);
        System.out.println("¿" + persona2.getNombre() + " es ahora mayor de edad? " + (persona2.esMayorDeEdad() ? "Sí" : "No"));
        persona2.mostrarDatos();
        System.out.println("\nVerificando apellidos paternos...");
        if (persona1.getPaterno().equals(persona2.getPaterno())) {
            System.out.println("Las dos personas tienen el mismo apellido paterno.");
        } else {
            System.out.println("Las dos personas tienen apellidos paternos diferentes.");
        }
    }
}